File: perl/classes/constructor_and_destructor.pl
#!/usr/bin/perl

package Item;

use strict;
use warnings;

sub new {
  print("Constructor is running...\n");
  my ($class, $args) = @_;
  my $self = bless {
  }, $class;
}

sub DESTROY {
  print("Destructor is running...\n");
  my $self = shift;
}

sub action {
  my $self = shift;
  print("Some action is performing...\n");
}

print("The object will be created now.\n");

my $thing = Item->new();
$thing->action();

print("The program will end now.\n");

File: php/classes/constructors_and_destructors/constructor_and_destructor.php
<?php
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

class Item {
  function __construct() {
    print("Constructor is running...\n");
  }

  function __destruct() {
    print("Destructor is running...\n");
  }

  function action() : void {
    print("Some action is performing...\n");
  }
}

print("The object will be created now.\n");

$thing = new Item();
$thing->action();

print("The program will end now.");
